src/protocolsupport/protocol/packet/middleimpl/clientbound/play/v_pe/BlockBreakAnimation.java

```diff
@@ -11,25 +11,24 @@
 
 public class BlockBreakAnimation extends MiddleBlockBreakAnimation {
 
-	private final TIntObjectHashMap<TimePle> animation = new TIntObjectHashMap<>();
+	private final TIntObjectHashMap<TimePle> animations = new TIntObjectHashMap<>();
 	
 	@Override
 	public RecyclableCollection<ClientBoundPacketData> toData(ProtocolVersion version) {
 		if (stage >= 1 && stage <= 9) {
-			if (!animation.containsKey(entityId)) {
-				animation.put(entityId, new TimePle(false, System.nanoTime(), stage));
+			TimePle animation = animations.get(entityId);
+			if (animation == null) {
+				animations.put(entityId, new TimePle(false, System.nanoTime(), stage));
 			} else {
-				if (!animation.get(entityId).getStarted() && stage > animation.get(entityId).getStartStage()) {
-					animation.get(entityId).setStarted(true);
+				if (!animation.getStarted() && stage > animation.getStartStage()) {
+					animations.get(entityId).setStarted(true);
```

Comment:
Indeed. I'll also adjust the calculation a bit. Seems that slow mining blocks can be calculated much more arurately and maybe shouldn't suffer under the extra constant.